internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/service/ScanServiceImpl.java

```diff
@@ -0,0 +1,328 @@
+
+    private static final String SCAN_REQUEST_FILE_MAP_ATTRIBUTE_NAME = "fileMap";
+    private static final String SCAN_REQUEST_PROPERTY_MAP_ATTRIBUTE_NAME = "propertyMap";
+    private static final String SCAN_REQUEST_SCAN_NAME_ATTRIBUTE_NAME = "scanName";
+    private static final String SCAN_REQUEST_SCAN_DESCRIPTION_ATTRIBUTE_NAME = "scanDescription";
+    private static final String SCAN_REQUEST_SCAN_TYPE_ATTRIBUTE_NAME = "scanType";
+    private static final String SCAN_REQUEST_PRODUCT_NAME_ATTRIBUTE_NAME = "productName";
+    private static final String SCAN_REQUEST_SCANNER_ID_ATTRIBUTE_NAME = "scannerId";
+    private static final String SCAN_ARTIFACT_DIRECTORY = "scans";
+
+    private static final Integer ARTIFACT_DOWNLOAD_CONNECTION_TIMEOUT = 10000;
+    private static final Integer ARTIFACT_DOWNLOAD_READ_TIMEOUT = 10000;
+
+    // temporary hold the scans till the pre scan tasks are completed.
+    private Map<String, Scan> waitingScans = new ConcurrentHashMap<>();
+
+    @Autowired
+    public ScanServiceImpl(LogService logService) {
+        this.logService = logService;
+    }
+
+    @Override
+    public Scan submitScan(Map<String, MultipartFile> fileMap, Map<String, String> parameterMap,
+                           String scanDirectory) {
+        Map<String, String> filesToBeDownloadedFromURL = new HashMap<>();
+        Map<String, String> storedFileMap = new HashMap<>();
+        File scanDirectoryLocation = null;
+
+        // Add scan to the waiting scans till the pre scan tasks are completed.
+        Scan scanWaiting = addScanToWaiting(parameterMap);
+        logService.insert(scanWaiting, LogType.INFO, "Scan is waiting to be submitted to scan manager API.");
+        try {
+            scanDirectoryLocation = new File(SCAN_ARTIFACT_DIRECTORY + File.separator + scanDirectory);
+            if (!scanDirectoryLocation.exists() && !scanDirectoryLocation.mkdirs()) {
+                throw new ScanManagerWebappException("Error occurred while creating the scan directory");
+            }
+
+            // store the uploaded files in a temp scan directory.
+            for (Map.Entry<String, MultipartFile> entry : fileMap.entrySet()) {
+                String unifiedFileName = entry.getKey() + "." +
+                        FilenameUtils.getExtension(entry.getValue().getOriginalFilename());
+                Path unifiedFilePath = Paths.get(scanDirectoryLocation.toPath().toString(), unifiedFileName);
+                writeToFile(entry.getValue(), unifiedFilePath);
+                storedFileMap.put(entry.getKey(), unifiedFilePath.toString());
+            }
+
+            Iterator<Map.Entry<String, String>> iter = parameterMap.entrySet().iterator();
+            while (iter.hasNext()) {
+                Map.Entry<String, String> param = iter.next();
+                if (param.getValue().isEmpty()) {
+                    iter.remove();
+                } else {
+                    if (param.getKey().contains("@byURL")) {
```

Comment:
Can we move `"@byURL"` to a constant. Since `@` is used as a separator in other locations too (line 132), it is better to keep it as a separate constant and `byURL` as another.